### ISSUE TITLE
Implement different examples to test decomposition algorithms

### ISSUE THREAD

**Issue by frapac** [exanauts/BlockNLPModels.jl#2]
I think it would be interesting to implement a bank of examples to test the different decomposition algorithms. Ideally, the examples would be generated such that
- they represent some kind of real-world problem ;
- we can parameterize their size (ie the number of blocks) ; 
- we can parameterize the couplings between the subproblems (that last point is more difficult though). 

For now, I have the following instances in mind:
- [ ] The single commodity network flow introduced in [Boyd's note](https://web.stanford.edu/class/ee364b/lectures/decomposition_notes.pdf)
- [ ] The LQR problem 
- [ ] the Network Utility Maximization and the Linear Network Flow problems introduced in [this paper](https://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.728.5548&rep=rep1&type=pdf) (randomly generated, graph structure)
- [ ] stochastic optimization instances (as those already implemented in [StructJuMP.jl](https://github.com/StructJuMP/StructJuMP.jl/tree/master/examples))
- [ ] the decentralized examples implemented in [ALADIN](https://github.com/alexe15/ALADIN.m/tree/master/examples)

Let me know if you have other examples in mind.

**Comment by grishabh147** [comment]
We can also implement some machine learning problems as almost all of them have a block structure. I would also like to test different algorithms on the inverse convex optimization problem (do not have a good reference yet).

**Comment by AnirudhSubramanyam** [comment]
I completely agree about having the ability to parameterize the size of individual blocks and the number of blocks. But shouldn't this be an issue in the `BlockNLPAlgorithms.jl` package?

**Comment by frapac** [comment]
> But shouldn't this be an issue in the BlockNLPAlgorithms.jl package?

you are totally right! I can transfer the issue as soon as BlockNLPAlgorithms.jl is created!
